components/locale/Cargo.toml

```diff
@@ -16,6 +16,7 @@ include = [
 
 [dependencies]
 tinystr = "0.3"
+serde = { version = "1.0", optional = true }
```

Comment:
Can you add the dependency when the feature is enabled, rather than making it an optional dependency?

Reply:
so, that's exactly what it does. Optional dependency means "in some scenarios `serde` may be used and then use 1.0".

When you do `cargo check` or `cargo build` it does not build `serde`, but if you do `cargo check --features serde` it will.

Think of it as if I added:

```
[features]
serde = ["serde"]
```

but I didn't have to because it matches the name 1-1.

Reply:
Sounds good; didn't know that's what `optional` meant!